draft-ietf-ppm-dap.md

```diff
@@ -730,37 +767,31 @@ Finally, the collector is configured with the HPKE secret key corresponding to
 
 ## Uploading Reports {#upload-flow}
 
-Clients periodically upload reports to the leader, which then distributes the
-individual shares to each helper.
+Clients periodically upload reports to the Leader, which then distributes the
+individual shares to each Helper. The upload sub-protocol involves the HPKE
+configurations ({{hpke-config-resource}}) and reports ({{report-resource}})
+resources.
 
-### HPKE Configuration Request {#hpke-config}
+### HPKE Configuration Resource {#hpke-config-resource}
 
 Before the client can upload its report to the leader, it must know the HPKE
 configuration of each aggregator. See {{compliance}} for information on HPKE
-algorithm choices.
-
-Clients retrieve the HPKE configuration from each aggregator by sending an HTTP
-GET request to `[aggregator]/hpke_config`, where `[aggregator]` is the
-aggregator's endpoint URL, obtained from the task parameters. Clients MAY
-specify a query parameter `task_id` when sending an HTTP GET request to
-`[aggregator]/hpke_config?task_id=[task-id]`, where `[task-id]` is the task ID
-obtained from the task parameters, encoded in Base 64 with URL and filename safe
-alphabet with no padding, as specified in sections 5 and 3.2 of {{!RFC4648}}. If
-the aggregator does not recognize the task ID, then it MUST abort with error
-`unrecognizedTask`.
+algorithm choices. Clients retrieve the HPKE configuration from each aggregator
+by accessing the HPKE configuration resource.
+
+#### Required methods
+
+##### GET `/hpke_config[?task_id={task-id}]`
```

Comment:
For consistency with other resources? Also, the return value is a list of HPKE configs.
```suggestion
##### GET `/hpke_configs[?task_id={task-id}]`
```

Reply:
Yes, excellent idea.